devbox.go

```diff
@@ -27,11 +27,12 @@ type Devbox interface {
 	Generate() error
 	GenerateDevcontainer(force bool) error
 	GenerateDockerfile(force bool) error
-	GenerateEnvrc(force bool, source string) error
+	CreateEnvrcFile(force bool) error
 	Info(pkg string, markdown bool) error
 	ListScripts() []string
 	PrintEnv(ctx context.Context, includeHooks bool) (string, error)
 	PrintGlobalList() error
+	PrintEnvrcContent(w io.Writer) error
```

Comment:
Should these print methods have similar parameters (`io.Writer`)?

Reply:
This function in particular should print to stdout. The others get their writer from devbox struct when we do `box.open()`.

Reply:
@hezhizhen I think that's a good point.

@mohsenari you are right that the other functions get their writer from box struct, but they only really make sense as stdout (and in that's how they are used).

In fact, `devbox.Open` is almost always called with stderr (except for list, generate, and info cases).

I think having `print` functions implicitly print to whatever is in devbox struct can be a bit error prone. Another way to think about it is that the `Writer` field in devbox is actually a logger (or just were we send diagnostics).

I would not suggest any action in this PR (I think the current signature is good). But maybe for the future:

* Default devbox.Writer to Stderr and remove from open() method.
* Refactor all print functions to take writer, or alternatively just have them all return the string.

Reply:
> Default devbox.Writer to Stderr and remove from open() method.

reminder: the `--quiet` flag is needs to be respected